sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/error.py

```diff
@@ -314,7 +314,8 @@ def __init__(self, status_code, status_description, **kwargs):
                 message += "\nDescription: {}".format(self.status_description.decode(encoding))
             except (TypeError, AttributeError):
                 message += "\nDescription: {}".format(self.status_description)
-        super(TokenAuthFailure, self).__init__(condition=ErrorCondition.ClientError, message=message)
+        condition = kwargs.get("condition") or ErrorCondition.ClientError
+        super(TokenAuthFailure, self).__init__(condition=condition, message=message)
```

Comment:
shouldn't be a breaking change since the value of condition is only changing for a scenario that was broken anyway